climada/hazard/tc_tracks_synth.py

```diff
@@ -834,6 +1720,1497 @@ def _check_decay_values_plot(x_val, v_lf, p_lf, v_rel, p_rel):
+        # apply decay (landfall) if track falls below the intensity of the final
+        # point in ibtracs
+        intensityrange_out_idx = _get_finalintensity_idx(track, original_track=tracks_list[i])
+        if intensityrange_out_idx < track.time.size:
+            if int(os.getenv('TRACKGEN_TEST_DECAY_BEYOND_ADVISORIES')):
+                if np.all(~sea_land) or np.where(sea_land)[0][0] > intensityrange_out_idx:
+                    sid_outside_intensity_range.append(track.sid)
+                    sea_land[intensityrange_out_idx] = True
+        if np.any(sea_land):
+            # apply landfall decay thereafter
+            sea_land_idx = np.where(sea_land)[0][0]
+            track['id_chunk'] = ('time', np.concatenate([
+                np.repeat([0], sea_land_idx),
+                np.repeat([-1], track.time.size - sea_land_idx)
+            ]).astype(int))
+            check_id_chunk(track['id_chunk'].values, track.sid, allow_missing=False)
+            # assign fake dist_since_lf
+            track['on_land'][sea_land_idx:] = True
+            track['dist_since_lf'] = ('time', climada.hazard.tc_tracks._dist_since_lf(track))
+            pcen_extend, v_extend = intensity_evolution_land(track,
+                                                             id_chunk=-1,
+                                                             v_rel=v_rel,
+                                                             p_rel=p_rel,
+                                                             s_rel=s_rel)
+            if v_extend is not None:
+                LOGGER.warning(
+                    f'Track extension not over owing to a landfall for {track.sid}. '
+                    f'Central pressure: {pcen_extend[-1]}, Max sustained wind: {v_extend[-1]}.'
+                )
+            elif pcen_extend is not None:
+                LOGGER.warning(
+                    f'Track extension not over owing to a landfall for {track.sid}. '
+                    f'Central pressure: {pcen_extend[-1]}.'
+                )
+            _estimate_params_track(track)
+        
+        # cutoff track end?
+        extended_cat = np.array([
+            climada.hazard.tc_tracks.set_category(
+                track['max_sustained_wind'].values[idx],
+                wind_unit=track.max_sustained_wind_unit
+            )
+            for idx in range(track.time.size)
+        ])
+
+        if np.any(extended_cat >= 0):  
+            # Kill the extended track 12 hours after dropping below Cat 1
+            buffer_frames = int(np.ceil(12/time_step_h))
+            cutoff_idx = min(np.where(extended_cat >= 0)[0][-1] + buffer_frames, extended_cat.size-1)
+            track = track.isel(time=slice(None, cutoff_idx))
+
+        # If the parent storm finished as extratropical, kill any extension frames at times beyond the original track end
+        original_size = tracks_list[i].time.size
+        if hasattr(track, 'nature'):
```

Comment:
@ChrisFairless Is my understanding correct that with this check, the model would work also with tracks without the 'nature' field but simply would not be apply to cut off the track in such a case?